build/AzurePipelinesTemplates/ProjectReunion-CreateNugetPackage-Job.yml

```diff
@@ -82,18 +85,28 @@ jobs:
   #       -BuildFlavor ${{ parameters.buildFlavor }}
 
   #   displayName: 'build-nupkg.ps1'
+  - task: PowerShell@2
+    displayName: UpdateFoundationNuSpecVersion
+    inputs:
+      targetType: 'inline'
+      script: |
+        $versionFinal = '${{ parameters.major }}.${{ parameters.minor }}.${{ parameters.patch }}.${{ parameters.versionDate }}.${{ parameters.versionCounter }}'
+        if(-not [String]::IsNullOrEmpty('${{ parameters.versionTag }}'))
+        {
+          $versionFinal = '${{ parameters.major }}.${{ parameters.minor }}.${{ parameters.patch }}-${{ parameters.versionTag }}.${{ parameters.versionDate }}.${{ parameters.versionCounter }}'
+        }
```

Comment:
So let's walk through this together with some test values. May want to have https://semver.org/ open as well.

Let's pretend we're releasing 1.0 to the world today, and your internal counter is zeroed.

$versionFinal becomes `1.0.0.20210318.0`. SemVer 2.0 states the version number should be in MAJOR.MINOR.PATCH form. So this is invalid.

It appears you're trying to add _build metadata_ here, so let's look at the spec:
> Build metadata MAY be denoted by appending a plus sign and a series of dot separated identifiers immediately following the patch or pre-release version.

Cool. Some examples:
* 1.0.0 (major release, no metadata, no pre-release identifiers)
* 1.0.0+0000.0 (major release with build metadata that does not influence decision making)
* 1.0.0-preview+20210318.1 (pre-release with build metadata [...])
* 1.0.0-preview+20210318.102 (pre-release with build metadata [...])

Reply:
(cont)

Be careful though. It may be tempting to represent everything as _build metadata_ but lets look at the spec again.
> Build metadata MUST be ignored when determining version precedence. Thus two versions that differ only in the build metadata, have the same precedence.

That means these are ranked the same (i.e. the client running .1 may not see .102 as different and not pull it down):
* 1.0.0-CI+20210318.1 (pre-release with build metadata [...])
* 1.0.0-CI+20210318.102 (pre-release with build metadata [...])

So you're going to need a hybrid approach here of pre-release identifiers and build metadata.